include accountId in cold delete kafka messages

welansari opened the pull request:
Cold delete Kafka messages need to contain the accountId like for every other cold operation.

Issue: BB-508

Commit:
include accountId in cold delete kafka messages

Issue: BB-508

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-508 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.6.38`

* `8.7.0`


Please check the `Fix Version/s` of BB-508, or the target
branch of this pull request.

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

francoisferrand commented on extensions/lifecycle/LifecycleQueuePopulator.js:
```diff
@@ -422,48 +422,60 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
-        const topic = `${coldStorageGCTopicPrefix}${locationName}`;
-        const key = `${entry.bucket}/${entry.key}`;
+        const ownerId = value['owner-id'];
+        this.vaultClientWrapper.getAccountId(ownerId, (err, accountId) => {
```
i don't think we can/want to afford the extra call to Vault for each message, when the value is actually not used?

→ One way would be to pass trough the `ownerId`, as fallback for `accountId` (both here & in Kafka), leaving all options open... but it may be a bit slow/unsafe to implement now?
→ Or, as a stop-gap measure, we may simply make the field required only for `archive` status messages

Commit:
make the accountId field optional for cold delete status messages

Making the accountId optional for delete operations as we might not
have it when the vault operation to get it fails. We still send the
message to Sorbet forwarder without the accountID as it is not used
and only kept for consistency.

Issue: BB-508

rachedbenmustapha commented on extensions/lifecycle/LifecycleQueuePopulator.js:
```diff
@@ -422,48 +423,63 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
-        const key = `${entry.bucket}/${entry.key}`;
+        const ownerId = value['owner-id'];
+        this.vaultClientWrapper.getAccountId(ownerId, (err, accountId) => {
+            // In case of faillure, we still send the delete request to Sorbet forwarder
```
```suggestion
            // In case of failure, we still send the delete request to Sorbet forwarder
```

rachedbenmustapha commented on extensions/lifecycle/LifecycleQueuePopulator.js:
```diff
@@ -422,48 +423,63 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
+        const ownerId = value['owner-id'];
+        this.vaultClientWrapper.getAccountId(ownerId, (err, accountId) => {
+            // In case of faillure, we still send the delete request to Sorbet forwarder
+            // without the accountId. The accountId is not used for delete operations. It is
+            // only included for consistency with other operations.
```
This may or may not remain true in the future, and this comment might become inaccurate if for some reason we start needing to update some other piece of metadata after GC.

Commit:
bump version 8.6.38

Issue: BB-508

welansari commented:
/approve

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-508` into
`development/8.6`
* [w/8.7/bugfix/BB-508](https://github.com/scality/backbeat/compare/development/8.7...w/8.7/bugfix/BB-508) will be merged into `development/8.7` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.6`

* :heavy_check_mark: `development/8.7`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.5`



Please check the status of the associated issue BB-508.

Goodbye kerkesni.




*The following options are set:* **approve**
